// babylonjs-viewer/configuration/types/default.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var babylonjs_viewer_assets_1 = require("babylonjs-viewer-assets");
var images = require("babylonjs-viewer-assets");
var renderOnlyDefault_1 = require("./renderOnlyDefault");
var helper_1 = require("../../helper");
/**
 * The default configuration of the viewer, including templates (canvas, overly, loading screen)
 * This configuration doesn't hold specific parameters, and only defines objects that are needed for the viewer to fully work correctly.
 */
exports.defaultConfiguration = helper_1.deepmerge(renderOnlyDefault_1.renderOnlyDefaultConfiguration, {
    version: "3.2.0-alpha4",
    templates: {
        main: {
            html: babylonjs_viewer_assets_1.defaultTemplate,
            params: {
                babylonFont: babylonjs_viewer_assets_1.babylonFont,
                noEscape: true
            }
        },
        fillContainer: {
            html: babylonjs_viewer_assets_1.fillContainer,
            params: {
                disable: false
            }
        },
        loadingScreen: {
            html: babylonjs_viewer_assets_1.loadingScreen,
            params: {
                backgroundColor: "#000000",
                loadingImage: images.loading,
                staticLoadingImage: images.staticLoading
            }
        },
        viewer: {
            html: babylonjs_viewer_assets_1.defaultViewer,
            params: {
                enableDragAndDrop: false
            }
        },
        navBar: {
            html: babylonjs_viewer_assets_1.navbar,
            params: {
                speedList: {
                    "0.5x": "0.5",
                    "1.0x": "1.0",
                    "1.5x": "1.5",
                    "2.0x": "2.0",
                },
                logoImage: images.babylonLogo,
                logoText: 'BabylonJS',
                logoLink: 'https://babylonjs.com',
                hideHelp: true,
                hideHd: true,
                hideVr: true,
                hidePrint: true,
                disableOnFullscreen: false,
                text: {
                    hdButton: "Toggle HD",
                    fullscreenButton: "Toggle Fullscreen",
                    helpButton: "Help",
                    vrButton: "Toggle VR",
                    printButton: "3D Print Object"
                }
            },
            events: {
                click: {
                    '.navbar-control': true
                },
                pointerdown: {
                    '.help-button': true
                },
                input: {
                    '.progress-wrapper': true
                },
                pointerup: {
                    '.progress-wrapper': true
                }
            }
        },
        overlay: {
            html: babylonjs_viewer_assets_1.overlay,
            params: {
                closeImage: images.close,
                closeText: 'Close'
            }
        },
        help: {
            html: babylonjs_viewer_assets_1.help
        },
        share: {
            html: babylonjs_viewer_assets_1.share
        },
        error: {
            html: babylonjs_viewer_assets_1.error
        }
    },
    camera: {
        behaviors: {
            autoRotate: {
                type: 0
            },
            framing: {
                type: 2,
                zoomOnBoundingInfo: true,
                zoomStopsAnimation: false
            },
            bouncing: {
                type: 1
            }
        }
    },
    skybox: {},
    ground: {
        receiveShadows: true
    },
    engine: {
        antialiasing: true
    },
    scene: {}
});
