// babylonjs-viewer/configuration/types/minimal.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var babylonjs_viewer_assets_1 = require("babylonjs-viewer-assets");
/**
 * The minimal configuration needed to make the viewer work.
 * Some functionalities might not work correctly (like fill-screen)
 */
exports.minimalConfiguration = {
    version: "0.1",
    templates: {
        main: {
            html: babylonjs_viewer_assets_1.defaultTemplate
        },
        fillContainer: {
            html: babylonjs_viewer_assets_1.fillContainer,
            params: {
                disable: false
            }
        },
        loadingScreen: {
            html: babylonjs_viewer_assets_1.loadingScreen,
            params: {
                backgroundColor: "#000000",
                loadingImage: babylonjs_viewer_assets_1.loading
            }
        },
        viewer: {
            html: babylonjs_viewer_assets_1.defaultViewer,
        },
        overlay: {
            html: babylonjs_viewer_assets_1.overlay,
            params: {
                closeImage: babylonjs_viewer_assets_1.close,
                closeText: 'Close'
            }
        },
        error: {
            html: babylonjs_viewer_assets_1.error
        }
    },
    engine: {
        antialiasing: true
    }
};
